noxfile.py

```diff
@@ -20,33 +20,42 @@
 ]
 
 # Other nox defaults
-nox.options.default_venv_backend = "venv"
+nox.options.default_venv_backend = "uv"
 nox.options.reuse_existing_virtualenvs = True
 
+# Note that everywhere we are installing our editables as (e.g.)
+# "./spawner[dev]" rather than "spawner[dev]"
+# This is a consequence of https://github.com/astral-sh/uv/issues/3994
+# which may be fixed by https://github.com/astral-sh/uv/pull/3995
+# but it's not clear that this isn't a consequence of PEP 508.
+#
+# In any event, it does no harm to make the fact that we're installing local
+# versions of these modules explicit.
+
 # pip-installable dependencies for development and documentation. This is not
 # used for pytest and typing, since it merges the controller, authenticator,
 # spawner, and inithome dependencies.
 PIP_DEPENDENCIES = [
-    ("--upgrade", "pip", "setuptools", "wheel"),
+    ("--upgrade", "uv", "pip", "setuptools", "wheel"),
```

Comment:
You can remove `pip`, `setuptools`, and `wheel` now that we're using `uv`. That will make this go faster.